src/fabtrimino.c

```diff
@@ -0,0 +1,250 @@
+#include "fabtrimino.h"
+
+struct square
+{
+    enum shape shape;
+};
+
+struct fabtrimino
+{
+    enum shape shape;
+    struct square*** matrix;
```

Comment:
Alright this is a 4x4 matrix of struct square objects. there's not real reason to allocate all of that on the heap as all sizes involved are static. Meaning that the following would work just fine:
```
struct square matrix[4][4];
```
Declaring your matrix like this will lay out 16 struct square object consecutively where they can be accessed like so:
```
trimino->matrix[1][1].shape;
```
Given that everything now has flat layout with a static shape, you no longer have to call calloc a bajillion times and you also avoid an ungodly amount of cache misses as your objects won't be scattered all over the heap (not entirely true as modern allocators are pretty good about maintaining locality). In any case, this is the simpler approach to creating static sized matrices.